master/buildbot/reporters/mail.py

```diff
@@ -43,6 +43,11 @@
 from buildbot.reporters.message import MessageFormatter as DefaultMessageFormatter
 from buildbot.util import service
 
+if PY3:
```

Comment:
Sorry, I was sure I did provide feedback on this.

The thing is that I am not sure what is the solution to this without actually spending lot of time trying the different options and eventually doing the work myself.

Doing import StringIO as BytesIO does not sound right, that I know for sure.

The current io.StringIO and io.BytesIO are not behaving the same in py2 and py3, and I think this is the core of the problem.

I dont think there is a nice solution to this problem.

So my input would be:
- stick with the future suggestion:
   only do from io import StringIO and from io import BytesIO.
- use BytesIO when we want bytes as *output*
- use StringIO when we want unicode as *output*
- pass the StringIO input through buildbot.util.ascii2unicode
- pass the BytesIO input through buildbot.util.as_utf8_bytes (to be created as @sa2ajj suggested)